### PR TITLE
FPE Mask and optimization

### PR DESCRIPTION
**Pull request by aehrath-gretel**
Added mask ability to FPE transformer and optimized an FPE helper function.

### CODE REVIEW

File: src/gretel_client/transformers/transformers/secure_fpe.py [hunk 1]
```diff
@@ -16,6 +16,8 @@ class SecureFpeConfig(RestoreTransformerConfig):
     radix: int = None
     type: type = None
     secret: str = None
+    mask: str = None
+    mask_fpe_char: chr = '0'
     float_precision: int = None
     aes_mode: Mode = Mode.CBC
```
Comment (johntmyers) [inline src/gretel_client/transformers/transformers/secure_fpe.py:22]: Please add **detailed** doc strings for this transformer config. It's a pretty complex one so we'll want to make sure each param is detailed. Also we can add simple in-line examples to doc strings too.

Example of doing that:

```
Args:
    foo: ...
    bar: ...

Example mask usage::

    sample python code here

```

Comment (johntmyers) [reply src/gretel_client/transformers/transformers/secure_fpe.py:22]: Another question, are `mask` and `float_precision` mutually exclusive? If both are not `None` should we throw a `ValueError` or something?

File: blueprints/simple_fpe.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
+"""
+Basic Format Preserving Encryption
+"""
+from gretel_client.transformers.transformers import SecureFpeConfig
+from gretel_client.transformers import DataPath, DataTransformPipeline
+
+xf = SecureFpeConfig(secret="2B7E151628AED2A6ABF7158809CF4F3CEF4359D8D580AA4F7F036D6F04FC6A94", radix=10,
+                     mask="1000 0000 0000 0000")
```
Comment (johntmyers) [inline blueprints/simple_fpe.py:8]: Question: the whitespaces in the mask, is that just cosmetic or does have it any affect on how things work? If just cosmetic (i.e. easier to read) maybe have an in-line comment that mentions that? I notice the example card below has whitespaces init so just wanted to know if the user should line up the whitespaces in the original data and the mask.

File: tests/src/gretel_client/unit/transformers/test_data_pipeline.py [hunk 3]
```diff
@@ -660,3 +660,60 @@ def test_record_fpe_base62():
...
     assert check == rec
+
+
+def test_record_fpe_mask():
+    rec = {'latitude': -70.783, 'longitude': -112.221, 'credit_card': '4123 5678 9123 4567', 'the_dude': 100000000,
+           'the_hotness': "convertme", "the_sci_notation": 1.23E-7}
+    cc_xf = [FormatConfig(pattern=r'\s+', replacement=''),
+             SecureFpeConfig(
+                 secret="2B7E151628AED2A6ABF7158809CF4F3CEF4359D8D580AA4F7F036D6F04FC6A94", radix=10,
+                 mask='*XXXXXXXXXXXXXXX', mask_fpe_char='X')]
```
Comment (johntmyers) [inline tests/src/gretel_client/unit/transformers/test_data_pipeline.py:671]: Is the leading `*` supported? I thought explicit masks were used

File: src/gretel_client/transformers/transformers/secure_fpe.py (hunk 1 above)
Comment (aehrath-gretel) [reply src/gretel_client/transformers/transformers/secure_fpe.py:22]: They are not. Float precision only works on values of type float not strings. Mask on the other hand only works on strings... resolving this cleanly does bring up the mask discussion again as at some point I was toying around with the idea to use the mask to specify the precision like "*.???" meaning preserve anything in front of the decimal point and up to 3 digits beyond the decimal point.

File: blueprints/simple_fpe.py (hunk 2 above)
Comment (aehrath-gretel) [reply blueprints/simple_fpe.py:8]: Yes cosmetic.. Anything that is not the mask_character get's left alone basically. I can add a comment.

File: tests/src/gretel_client/unit/transformers/test_data_pipeline.py (hunk 3 above)
Comment (aehrath-gretel) [reply tests/src/gretel_client/unit/transformers/test_data_pipeline.py:671]: It only checks for the character to be masked. Synonymous with the previous comment.

**Comment by johntmyers** [comment]
Closing this PR due to scope. As discussed we can specify sub-classes for specific partial value encryption. Starting with one for floats. We still need to figure out the interface design for partial strings.

**Pull request closed without merging**